### PR TITLE
fix pure hole child folding into a parent

### PR DESCRIPTION
**Pull request by jimengliu**
- replace seek_data/seek_hole using fiemap to loop through data extents
- add a test case that fold pure hole into parent

@yasker @sheng-liang

### CODE REVIEW

File: sparse/test/testutil_test.go [hunk 1]
```diff
@@ -125,7 +125,10 @@ func checkTestSparseFile(name string, layout []FileInterval) error {
 			if len(layoutActual) != 1 {
 				return errors.New(fmt.Sprint("hole check failure at", interval))
 			}
-			if layoutActual[0] != interval {
+			// actual hole must cover interval span
```
Comment (yasker) [inline sparse/test/testutil_test.go:128]: A question: why it's not exactly the same?

Comment (jimengliu) [inline sparse/test/testutil_test.go:128]: the hole/data we get back from fiemap may not start at the specified starting point, BeginOffset may be smaller than specified, EndOffset maybe larger than specified. So here we have DHH(data, hole, hole) blocks as the layout, we loop 2 times, but actual layout we get back is the same in these 2 times, it is always [1,3], not [1,2] first and [2,3] second.

The fix could either be merging the hole and don't loop 2 times, or change comparison statement as I did.

Comment (yasker) [inline sparse/test/testutil_test.go:128]: Is it valid case to have hole adjacent to another?

**Comment by yasker** [comment]
I will merge this one for now, just to unblock @sangeethah . But we need to do code cleanup/refactor for sparse-tools after this merged.

**Pull request merged**